pkg/controller/controller.go

```diff
@@ -46,6 +46,74 @@ const (
+	// when the pod ownership transfer from the controller to virt-hander succeeds.
+	SuccessfulHandOverPodReason = "SuccessfulHandOver"
+	// FailedDataVolumeImportReason is added in an event when a dynamically generated
+	// dataVolume reaches the failed status phase.
+	FailedDataVolumeImportReason = "FailedDataVolumeImport"
+	// FailedGuaranteePodResourcesReason is added in an event and in a vmi controller condition
+	// when a pod has been created without a Guaranteed resources.
+	FailedGuaranteePodResourcesReason = "FailedGuaranteeResources"
+	// FailedGatherhingClusterTopologyHints is added if the cluster topology hints can't be collected for a VMI by virt-controller
+	FailedGatherhingClusterTopologyHints = "FailedGatherhingClusterTopologyHints"
+	// FailedPvcNotFoundReason is added in an event
+	// when a PVC for a volume was not found.
+	FailedPvcNotFoundReason = "FailedPvcNotFound"
+	// SuccessfulMigrationReason is added when a migration attempt completes successfully
+	SuccessfulMigrationReason = "SuccessfulMigration"
+	// FailedMigrationReason is added when a migration attempt fails
+	FailedMigrationReason = "FailedMigration"
+	// SuccessfulAbortMigrationReason is added when an attempt to abort migration completes successfully
+	SuccessfulAbortMigrationReason = "SuccessfulAbortMigration"
+	// MigrationTargetPodUnschedulable is added a migration target pod enters Unschedulable phase
+	MigrationTargetPodUnschedulable = "migrationTargetPodUnschedulable"
+	// FailedAbortMigrationReason is added when an attempt to abort migration fails
+	FailedAbortMigrationReason = "FailedAbortMigration"
+	// MissingAttachmentPodReason is set when we have a hotplugged volume, but the attachment pod is missing
+	MissingAttachmentPodReason = "MissingAttachmentPod"
+	// PVCNotReadyReason is set when the PVC is not ready to be hot plugged.
+	PVCNotReadyReason = "PVCNotReady"
+	// FailedHotplugSyncReason is set when a hotplug specific failure occurs during sync
+	FailedHotplugSyncReason = "FailedHotplugSync"
+	// ErrImagePullReason is set when an error has occured while pulling an image for a containerDisk VM volume.
+	ErrImagePullReason = "ErrImagePull"
+	// ImagePullBackOffReason is set when an error has occured while pulling an image for a containerDisk VM volume,
+	// and that kubelet is backing off before retrying.
+	ImagePullBackOffReason = "ImagePullBackOff"
+	// NoSuitableNodesForHostModelMigration is set when a VMI with host-model CPU mode tries to migrate but no node
+	// is suitable for migration (since CPU model / required features are not supported)
+	NoSuitableNodesForHostModelMigration = "NoSuitableNodesForHostModelMigration"
+	// FailedPodPatchReason is set when a pod patch error occurs during sync
+	FailedPodPatchReason = "FailedPodPatch"
+	// MigrationBackoffReason is set when an error has occured while migrating
+	// and virt-controller is backing off before retrying.
+	MigrationBackoffReason = "MigrationBackoff"
+)
```

Comment:
IMO, it would be better to create a separate package just for functions and global variables. This way, you would reduce the risk of cyclic dependencies.

Reply:
In this case, I don't see it this way, since the `controller` pkg already exists and it fits perfectly to these constants (they are in the controller's scope). Creating a new pkg (I wouldn't even know what to call it honestly, since “controller” is already in use) gives no added value, as the controller package itself largely reduces the cyclic dependence, only importing external and KubeVirt API stuff.